lib/Dialect/TritonGPU/Transforms/OptimizeThreadLocality.cpp

```diff
@@ -305,6 +305,12 @@ class TritonGPUOptimizeThreadLocalityPass
     res.insert(res.begin() + index, static_cast<T>(value));
     return res;
   }
+  template <typename T>
+  SmallVector<T> insertValue(SmallVector<T> vec, unsigned index, int value) const {
```

Comment:
let's pass vec by reference to avoid an extra copy.

Reply:
Sure